aiida_common_workflows/workflows/eos.py

```diff
@@ -0,0 +1,158 @@
+            help='The list of scale factors at which the volume and total energy of the structure should be computed.')
+        spec.input('scale_count', valid_type=orm.Int, default=lambda: orm.Int(7), validator=validate_scale_count,
+            help='The number of points to compute for the equation of state.')
+        spec.input('scale_increment', valid_type=orm.Float, default=lambda: orm.Float(0.02),
+            validator=validate_scale_increment,
+            help='The relative difference between consecutive scaling factors.')
+        spec.input('generator_inputs', valid_type=orm.Dict, validator=validate_generator_inputs,
+            help='The inputs that will be passed to the inputs generator of the specified `sub_process`. It should '
+            'include all the inputs, excluding the `structure` and `previous_workchain`.')
+        spec.input_namespace('sub_process', dynamic=True, populate_defaults=False)
+        spec.input('sub_process_class', non_db=True, validator=validate_sub_process_class)
+        spec.inputs.validator = validate_inputs
+        spec.outline(
+            cls.run_init,
+            cls.run_eos,
+            cls.inspect_eos,
+        )
+        spec.output_namespace('structures', valid_type=orm.StructureData,
+            help='The relaxed structures at each scaling factor.')
+        spec.output_namespace('total_energies', valid_type=orm.Float,
+            help='The computed total energy of the relaxed structure at each scaling factor.')
+        spec.exit_code(400, 'ERROR_SUB_PROCESS_FAILED',
+            message='At least one of the `{cls}` sub processes did not finish successfully.')
+
+    def get_scale_factors(self):
+        """Return the list of scale factors."""
+        if 'scale_factors' in self.inputs:
+            return self.inputs.scale_factors
+
+        count = self.inputs.scale_count.value
+        increment = self.inputs.scale_increment.value
+        return [orm.Float(1 + i * increment - (count - 1) * increment / 2) for i in range(count)]
+
+    def get_sub_workchain_builder(self, scale_factor, previous_workchain=None):
+        """Return the builder for the relax workchain."""
+        structure = scale_structure(self.inputs.structure, scale_factor)
+        process_class = WorkflowFactory(self.inputs.sub_process_class)
+        generator_inputs = self.inputs.generator_inputs.get_dict()
+
+        builder = process_class.get_inputs_generator().get_builder(
+            structure,
+            calc_engines=generator_inputs['calc_engines'],
+            protocol=generator_inputs['protocol'],
+            relaxation_type=RelaxType(generator_inputs['relaxation_type']),
+            threshold_forces=generator_inputs.get('threshold_forces', None),
+            threshold_stress=generator_inputs.get('threshold_stress', None),
+            previous_workchain=previous_workchain
+        )
+        builder._update(**self.inputs.get('sub_process', {}))  # pylint: disable=protected-access
```

Comment:
Basically, the parameters passed to the namespace `sub_process` will override the parameters of the builder selected by the inputs generator. Right?

Reply:
> The implementation of Fleur still not working, but this is the general problem of having two codes, not of the EoS workchain. I guess @broeder-j needs to change the implementation at the end, right?
> All the rest is working well.

I have a separate branch where I have fixed some of it, but I am not done yet. The relax workchain runs. With the eos I had the problem of the structure input beeing structure.

Reply:
@broeder-j not sure if I understood, but the last modifications of Sebastiaan should fix this automatically. Now the `get_builder` of the inputs generator is called, and for sure it accepts `structure` as inputs in your implementation.

Reply:
Yes, although technically it is not necessary to keep this namespace, since a full set *will* be generated through the protocol  within the workchain, I decided to keep it in. As you have noticed, it allows to specify particular overrides. I am not yet sure how useful this will be, but on first glance, it actually deals with the problem I was describing. In this way, an "expert" can still use the `EosWorkChain` and override a particular input. They will have to know the exact expected namespace where to define the input of interest, but that is always the case. I am not yet sure if this will really be practical, but I think we can leave it in for now and see how it works out in practice.

Reply:
Yes, I think it is nice to allow expert users to modify the code. Good idea!